write-fonts/src/graph.rs

```diff
@@ -232,11 +241,33 @@ impl Graph {
         self.assign_spaces_hb();
         self.sort_shortest_distance();
 
-        let n_overflows = self.find_overflows().len();
-        if n_overflows > 0 {
-            log::info!("packing failed with {n_overflows} overflows");
+        if !self.has_overflows() {
+            return true;
         }
-        n_overflows == 0
+
+        // now isolate spaces in a loop, until there are no more left:
+        let overflows = loop {
+            let overflows = self.find_overflows();
+            if overflows.is_empty() {
+                // we're done
+                return true;
+            }
+            log::info!(
+                "failed with {} overflows, current size {}",
+                overflows.len(),
+                self.debug_len()
+            );
+            if !self.try_splitting_subgraphs(&overflows) {
+                log::info!("finished isolating all subgraphs without solution");
+                break overflows;
+            }
+            self.sort_shortest_distance();
+        };
+
+        assert!(!overflows.is_empty());
+        self.debug_overflows(&overflows);
+        //FIXME: return some kind of error here
```

Comment:
I suggest we might err on the side of issues over comments for this sort of thing, with possible exception for cases where we're going to fix it in O(days)

Reply:
Oops this comment is out of date; we check the return and convert to an error in the `dump_table` fn.